lightningd/offer.c

```diff
@@ -0,0 +1,209 @@
+#include <common/bolt12.h>
+#include <common/bolt12_merkle.h>
+#include <common/json_command.h>
+#include <common/json_helpers.h>
+#include <common/jsonrpc_errors.h>
+#include <common/param.h>
+#include <hsmd/hsmd_wiregen.h>
+#include <lightningd/jsonrpc.h>
+#include <lightningd/lightningd.h>
+#include <wallet/wallet.h>
+#include <wire/wire_sync.h>
+
+static void json_populate_offer(struct json_stream *response,
+				const struct sha256 *offer_id,
+				const char *b12,
+				const struct json_escape *label,
+				enum offer_status status)
+{
+	json_add_sha256(response, "offer_id", offer_id);
+	json_add_bool(response, "active", offer_status_active(status));
+	json_add_bool(response, "single_use", offer_status_single(status));
+	json_add_string(response, "bolt12", b12);
+	if (status == OFFER_USED)
+		json_add_bool(response, "used", true);
```

Comment:
Shouldn't we always include the `used` flag? Checking for existence and then checking the value on the client can be simplified if we do